src/h5web/visualizations/heatmap/AspectRatioToggler.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { FiToggleLeft, FiToggleRight } from 'react-icons/fi';
+import styles from './HeatmapToolbar.module.css';
+import { useHeatmapStore } from './store';
+
+function AspectRatioToggler(): JSX.Element {
+  const [keepAspectRatio, toggleAspectRatio] = useHeatmapStore(state => [
+    state.keepAspectRatio,
+    state.toggleAspectRatio,
+  ]);
+
+  return (
+    <button
+      className={styles.aspectRatioToggler}
+      type="button"
+      role="switch"
+      aria-checked={keepAspectRatio}
+      onClick={() => {
+        toggleAspectRatio();
+      }}
+    >
+      {keepAspectRatio ? <FiToggleRight /> : <FiToggleLeft />}
+      Keep aspect ratio
+    </button>
+  );
+}
+
+export default AspectRatioToggler;
```

Comment:
I put the aspect ratio toggler in this simple component to keep it as decoupled as possible. This way, we can move it around and inline it where necessary when the aspect of the toolbar is settled.